osu.Game/Screens/OsuScreen.cs

```diff
@@ -49,13 +51,15 @@ public WorkingBeatmap Beatmap
             }
             set
             {
-                beatmap.Value = value;
+                beatmap.Value = value ?? defaultBeatmap;

```

Comment:
I don't like properties where the getter returns something else than whatever has just been set. I'd prefer to throw when setting to null, and to use a method such as `SetDefaultBeatmap` to revert back to the default.

Alternatively, remove the setter completely and add a `SetBeatmap(WorkingBeatmap)` method which is documented to convert `null` to the default beatmap.